It appears some tests are failing due to a printing error for complex dtypes:
```
    def __call__(self, x):
>       r = self.real_format(x.real)
E       AttributeError: 'DeviceArray' object has no attribute 'real'
```
Should I just remove the JVP tests when the dtype is complex?

--- a/jax/lax_linalg.py
+++ b/jax/lax_linalg.py
@@ -479,19 +479,19 @@ def svd_jvp_rule(primals, tangents, full_matrices, compute_uv):
   k = s.shape[-1]
   Ut, V = np.conj(U).T, np.conj(Vt).T
   s_dim = s[..., None, :]
-  dS = Ut.dot(dA).dot(V)
+  dS = np.dot(np.dot(Ut, dA), V)
   ds = np.diag(dS)
   F = 1 / (np.square(s_dim) - np.square(s_dim.T) + np.eye(k)) - np.eye(k)
   dSS = s_dim * dS
   SdS = s_dim.T * dS
-  dU = U.dot(F * (dSS + dSS.T))
-  dV = V.dot(F * (SdS + SdS.T))
+  dU = np.dot(U, F * (dSS + dSS.T))
+  dV = np.dot(V, F * (SdS + SdS.T))
 
   m, n = A.shape[-2], A.shape[-1]
   if m > n:
-    dU = dU + (np.eye(m) - U.dot(Ut)).dot(dA).dot(V) / s_dim
+    dU = dU + np.dot(np.eye(m) - np.dot(U, Ut), np.dot(dA, V)) / s_dim
   if n > m:
-    dV = dV + (np.eye(n) - V.dot(Vt)).dot(dA.T).dot(U) / s_dim
+    dV = dV + np.dot(np.eye(n) - np.dot(V, Vt), np.dot(np.conj(dA).T, U)) / s_dim
   return core.pack((s, U, Vt)), core.pack((ds, dU, dV.T))
 
 def svd_cpu_translation_rule(c, operand, full_matrices, compute_uv):

--- a/jax/lax_linalg.py
+++ b/jax/lax_linalg.py
@@ -491,7 +491,7 @@ def svd_jvp_rule(primals, tangents, full_matrices, compute_uv):
   if m > n:
     dU = dU + (np.eye(m) - U.dot(Ut)).dot(dA).dot(V) / s_dim
   if n > m:
-    dV = dV + (np.eye(n) - V.dot(Vt)).dot(np.conj(dA).T).dot(U) / s_dim
+    dV = dV + (np.eye(n) - V.dot(Vt)).dot(dA.T).dot(U) / s_dim
   return core.pack((s, U, Vt)), core.pack((ds, dU, dV.T))
 
 def svd_cpu_translation_rule(c, operand, full_matrices, compute_uv):

--- a/tests/linalg_test.py
+++ b/tests/linalg_test.py
@@ -286,7 +286,7 @@ class NumpyLinalgTest(jtu.JaxTestCase):
                           args_maker, check_dtypes=True)
     if not full_matrices:
       svd = partial(np.linalg.svd, full_matrices=False)
-      jtu.check_jvp(svd, partial(jvp, svd), (a,))
+      jtu.check_jvp(svd, partial(jvp, svd), (a,), atol=1e-2 if FLAGS.jax_enable_x64 else jtu.ATOL)
 
   @parameterized.named_parameters(jtu.cases_from_list(
       {"testcase_name": "_shape={}_fullmatrices={}".format(

--- a/tests/linalg_test.py
+++ b/tests/linalg_test.py
@@ -286,7 +286,7 @@ class NumpyLinalgTest(jtu.JaxTestCase):
                           args_maker, check_dtypes=True)
     if not full_matrices:
       svd = partial(np.linalg.svd, full_matrices=False)
-      jtu.check_jvp(svd, partial(jvp, svd), (a,), atol=1e-2 if FLAGS.jax_enable_x64 else jtu.ATOL)
+      jtu.check_jvp(svd, partial(jvp, svd), (a,), atol=1e-1 if FLAGS.jax_enable_x64 else jtu.ATOL)
 
   @parameterized.named_parameters(jtu.cases_from_list(
       {"testcase_name": "_shape={}_fullmatrices={}".format(

